packages/zoe/test/unitTests/contracts/test-ownable-counter.js

```diff
@@ -0,0 +1,110 @@
+import { test } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+
+import path from 'path';
+
+import bundleSource from '@endo/bundle-source';
+import { E } from '@endo/eventual-send';
+import { GET_METHOD_NAMES } from '@endo/marshal';
+
+import { makeZoeForTest } from '../../../tools/setup-zoe.js';
+import { makeFakeVatAdmin } from '../../../tools/fakeVatAdmin.js';
+
+const filename = new URL(import.meta.url).pathname;
+const dirname = path.dirname(filename);
+
+const root = `${dirname}/ownable-counter.js`;
+
+test('zoe - ownable-counter contract', async t => {
+  const { admin: fakeVatAdmin, vatAdminState } = makeFakeVatAdmin();
+  const zoe = makeZoeForTest(fakeVatAdmin);
+  const invitationIssuer = await E(zoe).getInvitationIssuer();
+  const invitationBrand = await E(invitationIssuer).getBrand();
+
+  // Pack the contract.
+  const bundle = await bundleSource(root);
+  vatAdminState.installBundle('b1-ownable-counter', bundle);
+  const installation = await E(zoe).installBundleID('b1-ownable-counter');
+
+  const { creatorFacet: firstCounter, publicFacet: viewCounter } = await E(
+    zoe,
+  ).startInstance(
+    installation,
+    undefined,
+    undefined,
+    harden({
+      count: 3n,
+    }),
+    'c1-ownable-counter',
+  );
+
+  // the following tests could invoke `firstCounter` and `viewCounter`
+  // synchronously. But we don't in order to better model the user
+  // code that might be remote.
+
+  t.deepEqual(await E(firstCounter)[GET_METHOD_NAMES](), [
+    '__getInterfaceGuard__',
+    '__getMethodNames__',
+    'getInvitationCustomDetails',
+    'incr',
+    'makeTransferInvitation',
+  ]);
+
+  t.is(await E(firstCounter).incr(), 4n);
+  t.is(await E(viewCounter).view(), 4n);
+
+  t.deepEqual(await E(firstCounter).getInvitationCustomDetails(), {
```

Comment:
we rely on the ownable object to supply the custom details? not zoe?

Reply:
I think I see now... after `makeTransferInvitation`, we can ask zoe.

Reply:
Yup. Resolving.